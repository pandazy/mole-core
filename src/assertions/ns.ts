import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`assertions:${context}`);
}
