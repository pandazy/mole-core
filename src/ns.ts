export function ns(context: string): string {
  return `Mole:${context}`;
}
