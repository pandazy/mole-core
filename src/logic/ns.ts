import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`logic:${context}`);
}
