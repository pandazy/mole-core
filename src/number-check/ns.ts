import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`numericCheck:${context}`);
}
