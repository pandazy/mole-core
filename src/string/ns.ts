import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`string:${context}`);
}
