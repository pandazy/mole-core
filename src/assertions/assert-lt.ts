import { assertValue } from './assert-value';
import { isLt } from '../number-check/is-lt';

export const assertLt = (bar: number) => (context: string) => (value: number) =>
  assertValue<number>({
    isValid: isLt(bar),
    failureCause: `must be less than ${bar}`,
  })(context)(value);
