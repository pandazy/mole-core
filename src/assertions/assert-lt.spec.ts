import { assertLt } from './assert-lt';
import { ns } from './ns';

describe(ns('assertLt'), () => {
  it('should assert a value less than a number', () => {
    expect.assertions(1);

    const assertLt6 = assertLt(6)('test@arg');

    expect(() => assertLt6(7)).toThrowErrorMatchingSnapshot();
    assertLt6(3);
  });
});
