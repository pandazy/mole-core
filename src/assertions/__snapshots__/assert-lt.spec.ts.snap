// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mole:assertions:assertLt should assert a value less than a number 1`] = `
"
[context]: test@arg
[failure cause]: must be less than 6
[actual value]: 7
[actual value type]: Number"
`;
